### PR TITLE
Update Request Header

### PR DESCRIPTION
**Pull request by jumaallan**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Added security headers and interceptors

- Implemented timestamp and MAC generation

- Created security_info.json file

- Fixed deprecated feature flag property

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>lib.gradle.kts</strong><dd><code>Add SmileID security dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-9d62a200b35918c059dbb614af706a76e233386ff0bbdd8f7f45795c03f3e737">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>libs.versions.toml</strong><dd><code>Add SmileID security library dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td><strong>SmileID.kt</strong><dd><code>Add security interceptor to OkHttpClient</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-2f20b8276e08c468d9481c19866a6256498ded4517936a4fa68a2c696ebf1f63">+7/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Upload.kt</strong><dd><code>Add SecurityInfo data class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-137ac3c4a21f87d8931c1d4b62d79e5d923037f089841c43c6067047b2d36218">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>NetworkingUtil.kt</strong><dd><code>Add security_info.json to zip output</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-0a3922a1faa3013d250a9899cd8cd39236605a5571c94d34ef2ab5a778d51883">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileHeaderMetadataInterceptor.kt</strong><dd><code>Add timestamp header and keep-alive connection</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-bd6ab359bb4c8ba14d045147c93bb3a961b0101633094afd95582bad15ac06e7">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SmileIDService.kt</strong><dd><code>Add SmileIDSecurity annotation to API endpoints</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-31ced383f17efabef1c2ae957090b08345cee4065e2e385fcdadde2a2acdb3ed">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>FileUtils.kt</strong><dd><code>Add security info file creation utilities</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-1d536b452f414168ced6eddf1596e17f460a6355d98433927337b0c057856e48">+10/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SelfieViewModel.kt</strong><dd><code>Add timestamp generation for security info</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-53eee569c6733d62a661e2561d5e7af0daff5483c37973b79d154c70b3014937">+14/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>OrchestratedDocumentViewModel.kt</strong><dd><code>Add commented security info generation code</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-938003660e07ac447891364140d5e76f644af33d63d28c14e80e0785764d7d82">+15/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>sample.gradle.kts</strong><dd><code>Fix deprecated feature flag property</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-9b884f92ce13d0e2f784118cf6282e0de33bdc64a5986d1a051832b022f932d6">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>settings.gradle.kts</strong><dd><code>Add Maven snapshot repository</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/smileidentity/android/pull/536/files#diff-5625e3601fa0ad3a6a2824239e5a2fde71c149597d31394f9224a08c24be7b9d">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -38,6 +38,7 @@ okhttp = "4.12.0"
 play-services-mlkit-face-detection = "17.1.0"
 retrofit = "2.11.0"
 sentry = "7.20.1"
+smile = "1.0.0-20250401.125100-6"
```
Comment (robin-smileid) [inline gradle/libs.versions.toml:41]: Should we call this `smileid-security` ?

File: lib/src/main/java/com/smileidentity/SmileID.kt [hunk 2]
```diff
@@ -11,6 +11,7 @@ import com.google.android.gms.tasks.Tasks
 import com.google.mlkit.common.sdkinternal.MlKitContext
 import com.google.mlkit.vision.face.FaceDetection
 import com.serjltt.moshi.adapters.FallbackEnum
+import com.smileidentity.SmileID.initialize
```
Comment (robin-smileid) [inline lib/src/main/java/com/smileidentity/SmileID.kt:14]: Not sure we need this?

File: lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt [hunk 3]
```diff
@@ -56,6 +56,12 @@ fun UploadRequest.zip(): File {
     zipOutputStream.write(infoJson.toByteArray())
     zipOutputStream.closeEntry()
 
+    // Write security_info.json
+    zipOutputStream.putNextEntry(ZipEntry("security_info.json"))
+    val securityInfoJson = moshi.adapter(UploadRequest::class.java).toJson(uploadRequest)
+    zipOutputStream.write(securityInfoJson.toByteArray())
+    zipOutputStream.closeEntry()
```
[the remark this answers is not in the document]
Comment (robin-smileid) [reply lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt:63]: @jumaallan see the suggestion from PRfectionist.

File: lib/src/main/java/com/smileidentity/networking/SmileIDService.kt [hunk 4]
```diff
@@ -136,6 +139,7 @@ interface SmileIDService {
      * If the ID provider is unavailable, the response will be delivered to the callback URL once
      * the ID provider is available again.
      */
+    @SmileIDSecurity
     @POST("/v1/async_id_verification")
```
Comment (robin-smileid) [inline lib/src/main/java/com/smileidentity/networking/SmileIDService.kt:143]: We should add this to https://github.com/smileidentity/android/blob/d1065bc5ea943f0080a8c6c66106864c593fd36a/lib/src/main/java/com/smileidentity/networking/SmileIDService.kt#L133 as well

**Review by robin-smileid — COMMENTED** [review]
Left some comments. I'm also missing where you set the `SmileID-Request-Timestamp` header as in https://github.com/smileidentity/ios/blob/31361ff379daa82fab5f3e6575752fa8e4754316/Sources/SmileID/Classes/Networking/ServiceHeaderProvider.swift#L45.

File: lib/src/main/java/com/smileidentity/viewmodel/document/OrchestratedDocumentViewModel.kt [hunk 5]
```diff
@@ -156,6 +156,21 @@ internal abstract class OrchestratedDocumentViewModel<T : Parcelable>(
...
             )
 
+            // val timestamp = getCurrentIsoTimestamp()
+            // val mac = SmileIDCryptoManager.shared.sign(
+            //     timestamp = timestamp,
+            //     files = listOfNotNull(
+            //         frontImageInfo,
+            //         backImageInfo,
+            //         selfieImageInfo,
+            //     ) + livenessImageInfo,
```
[the remark this answers is not in the document]
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/viewmodel/document/OrchestratedDocumentViewModel.kt:166]: Needs to be tested with backend

**Comment by jumaallan** [comment]
> Left some comments. I'm also missing where you set the `SmileID-Request-Timestamp` header as in https://github.com/smileidentity/ios/blob/31361ff379daa82fab5f3e6575752fa8e4754316/Sources/SmileID/Classes/Networking/ServiceHeaderProvider.swift#L45.

It is there in SmileHeaderMetadata

File: lib/src/main/java/com/smileidentity/SmileID.kt (hunk 2 above)
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/SmileID.kt:14]: Will remove, was testing imports from security aar

File: lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt:63]: I wanted to write a new adapter actually for this when I can test with the api

File: settings.gradle.kts [hunk 6]
```diff
@@ -19,5 +19,6 @@ dependencyResolutionManagement {
     repositories {
         google()
         mavenCentral()
+        maven(url = "https://central.sonatype.com/repository/maven-snapshots/")
```
Comment (jumaallan) [inline settings.gradle.kts:22]: will remove this fyi (needed to get the snapshot, or we can leave it anyway, no harm)

**Review by JNdhlovu — COMMENTED** [review]
@jumaallan can we have a quick pair tomorrow on this so I can approve I think I am missing something 
`com.smileidentity.sample.debug: notify(1138, null, Notification(channel=chucker_transactions shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x110 color=0xff01579b actions=1 vis=PRIVATE semFlags=0x0 semPriority=0 semMissedCount=0)) as user
2025-05-08 22:56:49.377 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  <-- 401 https://api.smileidentity.com/v2/smart-selfie-enroll (2084ms)
2025-05-08 22:56:49.377 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  date: Thu, 08 May 2025 20:56:49 GMT
2025-05-08 22:56:49.377 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  content-type: application/json; charset=utf-8
2025-05-08 22:56:49.377 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  content-length: 76
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-requestid: cd93fc62-6ef7-46ac-b304-666531bf0167
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-x-amzn-requestid: 0903ca73-6f19-4237-a97c-9e710e319691
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  access-control-allow-origin: *
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  strict-transport-security: max-age=31536000; includeSubDomains
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-content-length: 76
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  allow: POST
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  access-control-allow-headers: Content-type
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-x-amzn-remapped-date: Thu, 08 May 2025 20:56:48 GMT
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-connection: keep-alive
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-cache: Error from cloudfront
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  via: 1.1 7a6352da3a4190b05338a7a6e5b1faee.cloudfront.net (CloudFront)
2025-05-08 22:56:49.378 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amz-cf-id: YhgiJ1a6T58fUssO95lFoon1hopa4194FiVkgDEDSAQgvRXdIgrZ_w==
2025-05-08 22:56:49.379 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amz-apigw-id: KREWgFLDCfMEtiA=
2025-05-08 22:56:49.379 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-x-amzn-remapped-content-length: 76
2025-05-08 22:56:49.379 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amz-cf-pop: CPT51-P1
2025-05-08 22:56:49.379 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-x-amzn-remapped-connection: keep-alive
2025-05-08 22:56:49.379 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  x-amzn-remapped-date: Thu, 08 May 2025 20:56:48 GMT
2025-05-08 22:56:49.386 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  {"code":"2205","error":"You are not authorized to do that.","success":false}
2025-05-08 22:56:49.386 32699-462   okhttp.OkHttpClient     com.smileidentity.sample.debug       I  <-- END HTTP (76-byte body)

Error during coroutine execution
                                                                                                    retrofit2.HttpException: HTTP 401 
                                                                                                    	at retrofit2.KotlinExtensions$await$2$2.onResponse(KotlinExtensions.kt:53)
                                                                                                    	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:164)
                                                                                                    	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
`

**Review by robin-smileid — DISMISSED** [review]
Code changes look fine to me. Tested all products and it works - before merge we just need to change the source_version.

File: lib/src/main/java/com/smileidentity/networking/SmileIDService.kt [hunk 7]
```diff
@@ -126,6 +131,8 @@ interface SmileIDService {
      * This will be done synchronously, and the result will be returned in the response. If the ID
      * provider is unavailable, the response will be an error.
      */
+    @SmileHeaderMetadata
+    @SmileIDSecurity
```
Comment (JNdhlovu) [inline lib/src/main/java/com/smileidentity/networking/SmileIDService.kt:135]: should we add 
 @SmileHeaderMetadata
    @SmileIDSecurity
    
    for doEnhancedKycAsync,/v1/job_status as well?

**Review by JNdhlovu — COMMENTED** [review]
left comments comments

File: lib/src/main/java/com/smileidentity/networking/SmileIDService.kt (hunk 7 above)
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/networking/SmileIDService.kt:135]: we did not add this on iOS too

**Comment by tobitech** [comment]
Tested all the jobs and worked fine.

File: lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt [hunk 8]
```diff
@@ -50,12 +53,28 @@ fun UploadRequest.zip(): File {
         )
     }
 
-    // Write info.json
-    zipOutputStream.putNextEntry(ZipEntry("info.json"))
+    // Create a temp file for info.json
+    val infoJsonFile = File.createTempFile("info", ".json")
     val infoJson = moshi.adapter(UploadRequest::class.java).toJson(uploadRequest)
+    infoJsonFile.writeText(infoJson)
+    infoJsonFile.deleteOnExit()
```
[the remark this answers is not in the document]
Comment (robin-smileid) [reply lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt:60]: We can keep this as improvement for next version. Thanks for spotting.

File: lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt [hunk 9]
```diff
@@ -127,3 +146,18 @@ fun File.asDocumentBackImage() = UploadImageInfo(
...
+
+/**
+ * Generates the security_info.json content for the zip file.
+ * This includes the mac and timestamp for verification.
+ *
+ * @param files The files from the request
+ */
+private fun createSecurityInfo(files: List<File>): SecurityInfoRequest {
+    val timestamp = getCurrentIsoTimestamp()
+    val signature = SmileIDCryptoManager.shared.sign(timestamp = timestamp, files = files)
```
[the remark this answers is not in the document]
Comment (jumaallan) [reply lib/src/main/java/com/smileidentity/networking/NetworkingUtil.kt:158]: This function does not throw any error
